trieste/models/gpflow/models.py

```diff
@@ -212,6 +214,70 @@ def evaluate_loss_of_model_parameters() -> tf.Tensor:
         multiple_assign(self.model, current_best_parameters)
 
 
+    def fantasized_predict_f(self, query_points: TensorType, fantasized_data: Dataset) -> tuple[TensorType, TensorType]:
```

Comment:
this is creating a lot of baggage in the `GaussianProcessRegression` wrapper that most acq fnc will not use, I would probably subclass it and add these methods in a new class

Reply:
as in creating a `FantasizeGPR`class? I was wondering if it made sense to also create a `FantasizeGPflowPredictor` in that case, any opinion?

Reply:
yes, `FantasizeGaussianProcessRegression` if we want to be consistent (names are getting long though :( )

if this would go beyond GPR, then yes, it would be good to standardise methods with a new interface like `FantasizeGPflowPredictor`, then it would be clear what methods a model would need to additionally implement

Reply:
If we subclass, the issue is that most users won't understand why they need to use this very specific wrapper if they want to use, say, the `tIMSE` acquisition?

Reply:
input to the timse acq function is a model, so perhaps then the right interface level would be `FantasizeTrainableProbabilisticModel` instead of `FantasizeGPflowPredictor` and then use it as a type of the input, that would be good enough indication what wrapper they need to use?

other than that so far we kinda concluded acq fncs will check that the models have the necessary methods, they won't keep a list of valid wrappers, but I agree its not an ideal solution - the user then has to dig through the models to find a wrapper with such a method 

keeping an explicit mapping between acq fncs and models has its own problem, perhaps we could probably come up with a utility function that scans wrappers in `models` package and lists all the wrappers with a given method?